src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -582,7 +582,11 @@ private BoundLambda ReallyInferReturnType(NamedTypeSymbol delegateType, Immutabl
             { WasCompilerGenerated = _unboundLambda.WasCompilerGenerated };
 
             // TODO: Should InferredReturnType.UseSiteDiagnostics be merged into BoundLambda.Diagnostics?
-            var returnType = inferredReturnType.Type ?? LambdaSymbol.InferenceFailureReturnType;
+            var returnType = inferredReturnType.Type;
+            if (returnType == null)
+            {
+                returnType = TypeSymbolWithAnnotations.CreateUnannotated(NonNullTypesFalseContext.Instance, LambdaSymbol.InferenceFailureReturnType);
```

Comment:
>NonNullTypesFalseContext.Instance [](start = 73, length = 33)

Same here

Reply:
Similar to `NoReturnExpression`, this value is a placeholder only, with only `TypeSymbol` used.

---
In reply to: [206364705](https://github.com/dotnet/roslyn/pull/28943#discussion_r206364705) [](ancestors = 206364705)

Reply:
Still, I think we should consider it. It probably doesn't make a big difference, but it will simplify the upcoming scrub (to make sure we pass proper contexts everywhere).

---
In reply to: [206540785](https://github.com/dotnet/roslyn/pull/28943#discussion_r206540785) [](ancestors = 206540785,206364705)